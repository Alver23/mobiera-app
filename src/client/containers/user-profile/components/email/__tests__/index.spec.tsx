// Dependencies
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { TouchableOpacity } from 'react-native';
import { Text } from 'react-native-elements';

// Under test file
import UserEmail from '../index';

describe('<UserEmail />', () => {
  let component: ShallowWrapper;
  const props = {
    email: 'fake@fake.com',
  };

  beforeEach(() => {
    component = shallow(<UserEmail {...props} />);
  });

  it('should save a snapshot of the component', () => {
    const onPress: any = component.find(TouchableOpacity).prop('onPress');
    if (onPress) {
      onPress();
    }
    expect(component).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const newProps = {
      ...props,
      onPress: jest.fn(),
    };
    component.setProps(newProps);
    const textLabel = component.find(Text).prop('children');
    const onPress: any = component.find(TouchableOpacity).prop('onPress');
    if (onPress) {
      onPress();
    }
    expect(textLabel).toBe(props.email);
    expect(newProps.onPress).toHaveBeenCalledWith(newProps.email);
  });
});
