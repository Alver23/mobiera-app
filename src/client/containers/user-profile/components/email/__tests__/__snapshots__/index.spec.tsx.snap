// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserEmail /> should save a snapshot of the component 1`] = `
<ForwardRef
  onPress={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Themed.Icon
      name="email"
      size={24}
      type="material"
    />
    <Themed.Text
      style={
        Object {
          "color": "#242424",
          "marginHorizontal": 4,
        }
      }
    >
      fake@fake.com
    </Themed.Text>
  </View>
</ForwardRef>
`;
